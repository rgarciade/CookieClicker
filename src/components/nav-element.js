import {LitElement,html,css} from "lit";
import {navStyle} from "./styles/nav-element-style.js";

export class NavElement extends LitElement {
    /**
     * @property {String} href - Link to navigate.
     */
    static get properties() {
        return {
            href: { type: String },
        };
    }
    static get styles() {
        return [navStyle];
    }

    constructor() {
        super();
    }
    _navigate(e) {
        e.preventDefault();
        const value = e.target.getAttribute('href')?? e.target.parentElement.getAttribute('href');
        this.dispatchEvent(new CustomEvent('router-navigate', {
            detail: value,
            bubbles: true,
            composed: true
        }));
    }
    render() {
        return html`
            <a href="${this.href}" @click="${this._navigate}"> <slot></slot></a>
        `;
    }
}

/**
 *
 * @param that - this ref to the element that dispatches the event.
 * @param href - The link to navigate.
 */
export const navigate = (elementRef,href) => {
    elementRef.dispatchEvent(new CustomEvent('router-navigate', {
        detail: href,
        bubbles: true,
        composed: true
    }));
}


window.customElements.define('nav-element', NavElement);
