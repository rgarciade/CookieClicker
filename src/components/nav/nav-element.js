import { LitElement, html } from 'lit';
import { navStyle } from './nav-element-style.js';

export class NavElement extends LitElement {
	/**
	 * @property {String} href - Link to navigate.
	 */
	static get properties() {
		return {
			href: { type: String },
		};
	}
	static get styles() {
		return [navStyle];
	}

	constructor() {
		super();
	}
	_navigate(e) {
		e.preventDefault();
		const value =
			e.target.getAttribute('href') ??
			e.target.parentElement.getAttribute('href');
		this.dispatchEvent(
			new CustomEvent('router-navigate', {
				detail: value,
				bubbles: true,
				composed: true,
			})
		);
	}
	render() {
		return html`
			<a href="${this.href}" @click="${this._navigate}"> <slot></slot></a>
		`;
	}
}

window.customElements.define('nav-element', NavElement);
