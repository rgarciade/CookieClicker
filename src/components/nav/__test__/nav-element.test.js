import { html, fixture } from '@open-wc/testing';
import '../nav-element.js';

import { expect } from '@jest/globals';
import { axe, toHaveNoViolations } from 'jest-axe';
import { cleanedLit } from '../../../../testFunctions/cleanedLit.js';
expect.extend(toHaveNoViolations);

describe('nav-element', () => {
	it('nav matches the snapshot', async () => {
		const el = await fixture(
			html`<navelement icon="person"
				><p>1</p>
				<p>2</p>
				<p>3</p></navelement
			>`
		);
		const dataCleaned = cleanedLit(el);
		expect(dataCleaned).toMatchSnapshot();
	});
	it('accessibility test', async () => {
		const el = await fixture(html`<nav-element icon="person"></nav-element>`);
		const results = await axe(el);
		expect(results).toHaveNoViolations();
	});
});
