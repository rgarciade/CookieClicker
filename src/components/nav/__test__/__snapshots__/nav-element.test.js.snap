// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nav-element nav matches the snapshot 1`] = `
"[36m<navelement[39m
  [33micon[39m=[32m"person"[39m
[36m>[39m
  [36m<p>[39m
    [0m1[0m
  [36m</p>[39m
  [36m<p>[39m
    [0m2[0m
  [36m</p>[39m
  [36m<p>[39m
    [0m3[0m
  [36m</p>[39m
[36m</navelement>[39m"
`;
